src/simpleNbox.cpp

```diff
@@ -998,4 +1132,201 @@ void SimpleNbox::set_c0(double newc0)
+void SimpleNbox::createBiome(const std::string& biome)
+{
+
+    H_LOG(logger, Logger::DEBUG) << "Creating new biome '" << biome << "'." << std::endl;
+
+    // Throw an error if the biome already exists
+    std::string errmsg = "Biome '" + biome + "' is already in `biome_list`.";
+    H_ASSERT(!has_biome( biome ), errmsg);
+
+    // Initialize new pools
+    veg_c[ biome ] = unitval(0, U_PGC);
+    add_biome_to_ts(veg_c_tv, biome, veg_c.at( biome ));
+    detritus_c[ biome ] = unitval(0, U_PGC);
+    add_biome_to_ts(detritus_c_tv, biome, detritus_c.at( biome ));
+    soil_c[ biome ] = unitval(0, U_PGC);
+    add_biome_to_ts(soil_c_tv, biome, soil_c.at( biome ));
+
+    npp_flux0[ biome ] = unitval(0, U_PGC_YR);
+
+    // Other defaults (these will be re-calculated later)
+    co2fert[ biome ] = 1.0;
+    tempfertd[ biome ] = 1.0;
+    add_biome_to_ts(tempfertd_tv, biome, 1.0);
+    tempferts[ biome ] = 1.0;
+    add_biome_to_ts(tempferts_tv, biome, 1.0);
+
+    std::string last_biome = biome_list.back();
+
+    // Set parameters to same as most recent biome
+    beta[ biome ] = beta[ last_biome ];
+    warmingfactor[ biome ] = warmingfactor[ last_biome ];
+    // TODO: Other parameters -- Q10, f_nppd, etc.
+
+    // Add to end of biome list
+    biome_list.push_back(biome);
+
+    H_LOG(logger, Logger::DEBUG) << "Finished creating biome '" << biome << "'." << std::endl;}
+
+// Delete a biome: Remove it from the `biome_list` and `erase` all of
+// the associated parameters.
+void SimpleNbox::deleteBiome(const std::string& biome) // Throw an error if the biome already exists
+{
+
+    H_LOG(logger, Logger::DEBUG) << "Deleting biome '" << biome << "'." << std::endl;
+
+    std::string errmsg = "Biome '" + biome + "' not found in `biome_list`.";
+    std::vector<std::string>::const_iterator i_biome = std::find(biome_list.begin(), biome_list.end(), biome);
+    H_ASSERT(has_biome( biome ), errmsg);
+
+    // Erase all values associated with the biome:
+    // Parameters
+    beta.erase( biome );
+    warmingfactor.erase( biome );
+
+    // C pools
+    veg_c.erase( biome );
+    remove_biome_from_ts(veg_c_tv, biome);
+    detritus_c.erase( biome );
+    remove_biome_from_ts(detritus_c_tv, biome);
+    soil_c.erase( biome );
+    remove_biome_from_ts(soil_c_tv, biome);
+
+    // Others
+    npp_flux0.erase( biome );
+    tempfertd.erase( biome );
+    remove_biome_from_ts( tempfertd_tv, biome );
+    tempferts.erase( biome );
+    remove_biome_from_ts( tempferts_tv, biome );
+    co2fert.erase( biome );
+
+    // Remove from `biome_list`
+    biome_list.erase( i_biome );
+
+    H_LOG(logger, Logger::DEBUG) << "Finished deleting biome '" << biome << ",." << std::endl;
+
+}
+
+// Create a new biome called `newname`, transfer all of the parameters
+// and pools from `oldname`, and delete `oldname`. Note that the new
+// biome and all of its associated values will be at the end of their
+// corresponding vectors.
```

Comment:
The associated values are in maps, anyhow, aren't they?  In that case, the last added value will generally not be at the end (and I'm not sure those structures even provide a `back` method in any case).

Reply:
Ah, good catch, thanks! This is technically still true of `biome_list` (see my [comment above](https://github.com/JGCRI/hector/pull/317#discussion_r301640644)), but I'll remove the part about the values.

Reply:
Addressed in dc8f5f0c6186ecfbe51aa1d0b8c97ed34fa54150.